src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,46 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
     public void getStatistic(String fromFileName, String toFileName) {
+        int supply = 0;
+        int buy = 0;
+
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(",");
+                if (parts.length != 2) {
+                    continue;
+                }
+                String operation = parts[0].trim();
+                int amount = Integer.parseInt(parts[1].trim());
+
+                if ("supply".equals(operation)) {
+                    supply += amount;
+                } else if ("buy".equals(operation)) {
+                    buy += amount;
+                }
```

Lines 24-28: The code ignores unknown operation types silently (no explicit handling). This violates: "Handle unknown operation types explicitly: either ignore or throw a RuntimeException; behavior must be consistent and justified in code." Decide and implement one behavior (throw or skip) and document it in code; failing fast is recommended for unexpected input .